src/Categories/Morphism/Extremal/Properties.agda

```diff
@@ -0,0 +1,221 @@
+-- Extremal Monomorphisms
+
+RegularMono⇒IsExtremalMono : ∀ {A B} {f : A ⇒ B} (m : RegularMono f) → IsExtremalMono (RegularMono⇒Mono m)
+RegularMono⇒IsExtremalMono {A = A} {B = B} {f = f} regular {X = X} {g = k} {i = i} i-epi f≈k∘i =
+  record
+    { inv = i⁻¹
+    ; iso = record
+      { isoˡ = isoˡ
+      ; isoʳ = isoʳ
+      }
+    }
+    where
+      open RegularMono regular renaming (g to r; h to s)
+      open IsEqualizer equalizer
+
+      i⁻¹ : X ⇒ A
+      i⁻¹ = equalize $ i-epi (s ∘ k) (r ∘ k) $ begin
+        (s ∘ k) ∘ i ≈⟨ pullʳ (⟺ f≈k∘i) ⟩
+        s ∘ f       ≈⟨ equality ⟩
+        r ∘ f       ≈⟨ pushʳ f≈k∘i ⟩
+        (r ∘ k) ∘ i ∎
+
+      isoˡ : i⁻¹ ∘ i ≈ id
+      isoˡ = RegularMono⇒Mono regular (i⁻¹ ∘ i) id $ begin
+        f ∘ i⁻¹ ∘ i ≈⟨ pullˡ (⟺ universal) ⟩
+        k ∘ i       ≈˘⟨ f≈k∘i ⟩
+        f           ≈˘⟨ identityʳ ⟩
+        f ∘ id      ∎
+
+      isoʳ : i ∘ i⁻¹ ≈ id
+      isoʳ = i-epi (i ∘ i⁻¹) id $ begin
+        (i ∘ i⁻¹) ∘ i ≈⟨ cancelʳ isoˡ ⟩
+        i             ≈˘⟨ identityˡ ⟩
+        id ∘ i        ∎
+
+RegularMono⇒ExtremalMono : RegularMono f → ExtremalMono f
+RegularMono⇒ExtremalMono m = record
+  { mono = RegularMono⇒Mono m
+  ; extremal = RegularMono⇒IsExtremalMono m
+  }
+
+ExtremalMono-∘₂ : ExtremalMono (f ∘ g) → ExtremalMono g
+ExtremalMono-∘₂ fg-extremal = record
+  { mono = Mono-∘₂ mono
+  ; extremal = λ i-epi eq → extremal i-epi (pushʳ eq)
+  }
+  where
+    open ExtremalMono fg-extremal
+
+ExtremalMono-∘ : ∀ {A B C} {f : B ⇒ C} {g : A ⇒ B} → RegularMono f → ExtremalMono g → ExtremalMono (f ∘ g)
+ExtremalMono-∘ {A = A} {B = B} {f = f} {g = g} f-regular g-extremal = record
+  { mono = fg-mono
+  ; extremal = fg-extremal
+  }
+  where
+    module f-regular = RegularMono f-regular
+    open IsEqualizer f-regular.equalizer
+    module g-extremal = ExtremalMono g-extremal
+
+    f-mono : Mono f
+    f-mono = (RegularMono⇒Mono f-regular)
+
+    fg-mono : Mono (f ∘ g)
+    fg-mono = Mono-∘ f-mono g-extremal.mono
+
+    fg-extremal : IsExtremalMono fg-mono
+    fg-extremal {X = X} {g = h} {i = e} e-epi f∘g≈h∘e =
+      g-extremal.extremal e-epi $ f-mono _ _ $ begin
+      f ∘ g     ≈⟨ f∘g≈h∘e ⟩
+      h ∘ e     ≈⟨ pushˡ universal ⟩
+      f ∘ k ∘ e ∎
+      where
+        k : X ⇒ B
+        k = equalize $ e-epi _ _ $ begin
+          (f-regular.h ∘ h) ∘ e ≈⟨ extendˡ (⟺ f∘g≈h∘e) ⟩
+          (f-regular.h ∘ f) ∘ g ≈⟨ equality ⟩∘⟨refl ⟩
+          (f-regular.g ∘ f) ∘ g ≈⟨ extendˡ f∘g≈h∘e ⟩
+          (f-regular.g ∘ h) ∘ e ∎
+
+ExtremalMono+Epi⇒IsIso : ExtremalMono f → Epi f → IsIso f
+ExtremalMono+Epi⇒IsIso {f = f} f-extremal f-epi = extremal f-epi (⟺ identityˡ)
+  where
+    open ExtremalMono f-extremal
+
+--------------------------------------------------------------------------------
+-- Extremal Morphisms in Balanced Categories
+-- https://ncatlab.org/nlab/show/balanced+category
+
+module _ (balanced : ∀ {A B} {f : A ⇒ B} → Mono f → Epi f → IsIso f) where
```

Comment:
At some point it might be good to start a module for balanced categories, at which point this should be refactored.